### DISCUSSION TITLE
spell.c: apostrophe as word boundary (for french)

### DISCUSSION THREAD

**Discussion by thjbdvlt** [neovim/neovim#27996] [Q&A]
hello! 

i've searched for a way to tell the neovim spell to process apostrophes just like spaces (as word boundaries), but i couldn't find any way to do it. so i'm trying to modify the `spell.c` file but i cannot manage to make it work. has someone any idea which part i should modify?

because in french, it's how apostrophe is used. for example, the word 'arriver' (to come) (or any verb beginning with a voyel) can be used like this: "j'arrive" (i come), "n'arriver" (not to come), "t'arriver" (to come to you), "m'arriver" (to come to me), "d'arriver" (to come), "qu'arriver" (just to come), and so on. the pattern _word + apostrophe + word_ is very very very common (i'd say 90% of the sentences i write). and i don't want to add theses patterns in my spell files, because it would massively increase their size (and because it's against the french grammar). in the other hand, there is only one word containing an apostrophe, it's "aujourd'hui" (and its actually many words joined into one).

of course, it's only for my personal use, i don't know very well the syntax of other natural languages so i don't know if it's a french-only case (but if it's usefull, i can try to do something).

**Comment by Phantasimay** [comment]
what spell do you use? plugin or native spell? what exacly your problem? is it about false positive spell to detect your personal word? so i can try to reproduct in my machine.

FYI:
if you just want to detect apostrophe like space you can use `isfname`:

```
	setlocal isfname+=32,[,',?,`,;,&  "all char as literal
	setlocal isfname-=#	          "all char as space(non literal)
	setlocal path+=$WIKI_DIR/         "target dir
	setlocal suffixesadd+=.adoc       "target file extension
```

`:set isfname` to check default isfname

**Reply by thjbdvlt** [reply]
thank's for your answer! :) 
i use the native neovim spell, and my problem is wrong negatives. for example i've added the word _auteurice_. and _l'auteurice_, _d'auteurice_, _qu'auteurice_ or _presqu'auteurice_ are considered wrong (hence underlined). but there is no reason for that because in a form like _d'auteurice_ there are two words, not one. it should be the same as _d.auteurice_. for verbs it becomes unmanageable because for verbs there is like 17 forms like this.

i've tried to set the isfname, but it does not change anything, words are still underlined as wrong.

**Reply by Phantasimay** [reply]
as far as i know native spellcheck
 
>  Vim only checks words for spelling, there is no grammar check.

so what we can do are add your personal grammer with `zg` it will save word undercusor to your personal grammer in `$HOME/.config/spell/fr.add`
hope it will satified you.

**Reply by thjbdvlt** [reply]
i would like to avoid `zg` every combination, as the number is really huge (it's why i want neovim not to consider apostophe as a word character). but it seems i have to rewrite .aff and .dic files and recompile with mkspell.

**Comment by thjbdvlt** [comment]
if anyone someday looks for the same thing: i wrote an alternative spell file (that doesnt consider apostrophe/hyphen as word character) and i put i [there](https://github.com/thjbdvlt/spell-fr.vim). so i close the discussion :)
